File: passwords.cxx
#include <iostream>
#include <string>

int main() {
    std::string input;
    while(std::getline(std::cin, input)) {
        size_t password_count = 0;
        bool up = false, low = false, dig = false;
        for(int i = 0; i != input.length(); ++i) {
            if(isupper(input[i]))
                up = true;
            else if(islower(input[i]))
                low = true;
            else if(isdigit(input[i]))
                dig = true;
            if(up && low && dig) {
                up = low = dig = false;
                ++password_count;
            }
        }
        std::cout << password_count << std::endl;   
    }
}

File: passwords.rs
use std::io::{self, BufRead};

fn main() {
   //let mut input = String::new();
   let stdin = io::stdin();
   for line in stdin.lock().lines() {
        let input = line.unwrap();
        let mut password_count = 0;
        let mut up = false;
        let mut low = false;
        let mut dig = false;

        for i in 0..input.len() {
            if input.chars().nth(i).unwrap().is_uppercase() {
                up = true;
            }
            else if input.chars().nth(i).unwrap().is_lowercase() {
                low = true;
            }
            else if input.chars().nth(i).unwrap().is_digit(10) {
                dig = true;
            }
            if up && low && dig {
                up = false;
                low = false;
                dig = false;
                password_count += 1;
            }
        }
        println!("{}", password_count);
    }
}
